### ISSUE TITLE
Error due to hardcoded home directory - /home/sascha/

### ISSUE THREAD

**Issue by DominicMe** [hyperion-project/hyperion#788]
I am getting the following error:

`Unable to open RS232 device (IO Exception (2): No such file or directory, file /home/sascha/hyperion/dependencies/build/serial/src/impl/unix.cc, line 100.)`

There is no user named sascha so I can only assume it was accidentally hardcoded by the dev using username sascha.

**Comment by Paulchen-Panther** [comment]
Brindosch is Sascha

**Comment by penfold42** [comment]
The no such file or directory refers to the serial device name you (wrongly) specified in the configuration.

That error occurred in unix.cc line 100

**Comment by DominicMe** [comment]
It was correctly specified but disconnected so same result I guess. That is not the issue here though. Issue that the home directory is hard coded to that of the developer who wrote the code. Also even if it pointed to correct home directory it still wrong as the file in question does not reside in the home directory at all and why would it since hyperion needs to run as root.

**Comment by penfold42** [comment]
There’s nothing hard coded - that message refers to the directory path used when compiling.

That path has _nothing_ to do with Hyperion at runtime.

Back to your actual problem: are you seeing USB device disconnects in dmesg output ?

**Comment by Paulchen-Panther** [comment]
total nonsense. search the repository and you will not find the name sascha.

**Comment by DominicMe** [comment]
There is no such directory on my system, so where did this error come from then?? There was never any directory named "sacha" on my system so the error could only have come from hyperion.

Maybe some files were leftover from auto install script so maybe it uses a different version? I would have assume it would use the latest official release...

**Comment by Paulchen-Panther** [comment]
Get it with git clone and compile it and you will see there is no more sascha

**Comment by penfold42** [comment]
You do realise you’re arguing with 2 of the developers right ?

**Comment by DominicMe** [comment]
Yes, but the only time I mentioned a dev was relating to code written presumably by Sascha/Brindosch.

I can confirm that it's an issue with the auto install script and not the latest git code. Vast majority will use the script so it should be updated to use latest release which doesn't have this bug. I don't know if there's maybe a disconnect between git repo and the website...

**Comment by deisi** [comment]
I was also surprised by this. In my log I found:

```
Apr 11 12:35:05 serv hyperiond[2934]: Device blocked for 3000 ms
Apr 11 12:35:08 serv hyperiond[2934]: Device unblocked
Apr 11 12:35:08 serv hyperiond[2934]: Opening UART: /dev/hyperion
Apr 11 12:35:08 serv hyperiond[2934]: Unable to open RS232 device (IO Exception: ::tcgetattr, file /home/sascha/hyperion/dependencies/build/serial/src/impl/unix.cc, line 119.)
```

Now hyperion sometimes works, but sometimes also not. I don't know if this is related.

> total nonsense. search the repository and you will not find the name sascha.

I don't want to contribute too much to the semantics weather this is hard coded or not, but I think we can in general agree that `Unable to open RS232 device (IO Exception: ::tcgetattr, file /home/sascha/hyperion/dependencies/build/serial/src/impl/unix.cc, line 119.)`
Is not a very good error message. Regardless of the cause. 

However I have a little funfact here. you can actually find the word `sasha` in the repository.  Do a `cat -v /usr/bin/hyperiond  | grep sascha` and you find the string within the binary itself ;). So given that the release binary is part of the repository, the word `sascha` is technically in.
